### ISSUE TITLE
Sometimes the way of XLA updating tensors may be confusing.

### ISSUE THREAD

**Issue by cicirori** [pytorch/xla#3529]
## 🐛 Bug

```
import torch

import torch_xla
import torch_xla.core.xla_model as xm


a = torch.tensor(1, device=xm.xla_device())
b = torch.tensor(1, device=xm.xla_device())
c = torch.tensor(1, device=xm.xla_device())

xm.mark_step()
e = a * c
d = e * a 
b += d
for i in range(10):
    print(b.item())

# output : 2 3 4 5 6 7 8 9 10 11
```

## Expected behavior

Always print 2

## Additional context

I can understand that the reason for this comes from the fact that the tensor.item() call does not clear the IR grpah of the tensor.

But this is a common code pattern for algorithm engineers to record loss and output, and xm.add_step_closure might be the answer to this problem. But it was still very annoying at first (I wasn't aware that a tensor had been accidentally updated multiple times, only to get a very poor loss and make me spend a lot of time on some other factors that might affect convergence).

Would it be possible to mention this more explicitly in the documentation ([maybe here](https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md#xla-tensor-quirks)), or to throw a warning when tracing the corresponding pattern.
